File: Data Structures & Algorithms/reorder-linked-list/submission-0.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */

class Solution {
public:
    void reorderList(ListNode* head) {
        stack<ListNode*> st;
        ListNode* front = head;
        while(front!=NULL){
            st.push(front);
            front=front->next;
        }
        ListNode* temp = head;
        int half_size = st.size() / 2;
        for(int i = 0; i < half_size; i++){
            ListNode* fron = temp->next;
            temp->next = st.top();
            st.pop();
            temp = temp->next;
            temp->next=fron;
            temp = temp->next;
           }
           if(temp!=NULL){
            temp->next = NULL;
           }
    }
};

File: Data Structures & Algorithms/linked-list-cycle-detection/submission-0.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next

class Solution:
    def hasCycle(self, head: Optional[ListNode]) -> bool:
        mpp = {}
        temp = head
        while temp != None:
            if temp not in mpp:
                mpp[temp] = 0
            mpp[temp] += 1
            if mpp[temp] == 2:
                return True
            temp = temp.next
        return False
        